ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -336,6 +336,63 @@ function upgrade() {
         clear_upgrade
         check_upgrade_status "${cluster_src_ver}"
     fi
+    if check_ota_case_enabled "OCP-25473"; then
+        local retry=0 previous_image output PREVIOUS_MINOR_VERSION
+        previous_image="$(< "${SHARED_DIR}/upgrade-edge")"
+        if [[ -n "${previous_image:-}" ]]; then
+            run_command "oc adm upgrade --to-image=${previous_image} --allow-explicit-upgrade --force=${FORCE_UPDATE}"
+        else
+            echo "Error: The previous image is not given, break the job"
+            exit 1
+        fi
+        
+        echo "OCP-25473: Waiting for the first time upgrade get to start"
+        while [[ "$(oc get clusterversion version -o json|jq '.status.conditions[]|select(.type=="Progressing")'|jq -r '.status')" != 'True' ]]; do
+            if [[ retry -ge 5 ]]; then
+                echo "Error: OCP-25473: First time upgrade not started, break the job"
+                exit 1
+            fi
+            retry=$(( retry+1 ))
+            sleep 1m
+        done
+        echo "OCP-25473: Upgrade is processing"
+
+        echo "OCP-25473: Upgrade again without --allow-upgrade-with-warnings"
+        oc adm upgrade --to-image=${TARGET} --allow-explicit-upgrade --force=${FORCE_UPDATE} > ${ARTIFACT_DIR}/ota_25473_result 2>&1 || true
+        output="$(cat ${ARTIFACT_DIR}/ota_25473_result)"
+        if [[ ! "${output}" =~ "error: the cluster is already upgrading" ]] || [[ ! "${output}" =~ "If you want to upgrade anyway, use --allow-upgrade-with-warnings" ]]; then
+            echo "Error: OCP-25473:  Re-upgrade should not start and raise error when there is already an upgrade is processing"
+            echo "The output of 'oc adm upgrade':"
+            echo $output
+            exit 1
+        fi
+        
+        echo "OCP-25473: Upgrade again with --allow-upgrade-with-warnings"
+        # oc adm upgrade --to-image=${TARGET} --allow-explicit-upgrade --allow-upgrade-with-warnings=true --force=${FORCE_UPDATE} > ${ARTIFACT_DIR}/ota_25473_result 2>&1 || true
+        # output="$(cat ${ARTIFACT_DIR}/ota_25473_result)"
+        PREVIOUS_MINOR_VERSION=$(oc get clusterversion version -ojson | jq -r '.status.desired.version' | cut -f2 -d.)
+        run_command "oc adm upgrade --to-image=${TARGET} --allow-explicit-upgrade --allow-upgrade-with-warnings=true --force=${FORCE_UPDATE}"
+        if [[ "$PREVIOUS_MINOR_VERSION" == "$TARGET_MINOR_VERSION" ]]; then
```

Comment:
@ @evakhoni  As designed by https://issues.redhat.com/browse/OTA-861 , when "$PREVIOUS_MINOR_VERSION" == "$TARGET_MINOR_VERSION", no matter this is a y stream or a z stream upgrade, we should allow it to continue.